[ADD] Upgrade imgui to v1.67

Galaco opened the pull request:
Updated included imgui source to release v1.67.

Only known breaking change to existing bindings were removal of:
* ImFontAtlas::GetGlyphRangesChinese ( ) 

And replaced with:
* ImFontAtlas::GetGlyphRangesChineseFull ( ) 
* ImFontAtlas::GetGlyphRangesChineseSimplifiedCommon ( ) 

Naturally bindings have been updated accordingly.

dertseha commented on FontAtlas.go:
```diff
@@ -39,9 +39,14 @@ func (atlas FontAtlas) GlyphRangesJapanese() GlyphRanges {
-// GlyphRangesChinese describes Default + Japanese + full set of about 21000 CJK Unified Ideographs.
-func (atlas FontAtlas) GlyphRangesChinese() GlyphRanges {
-	return GlyphRanges(C.iggGetGlyphRangesChinese(atlas.handle()))
+// GlyphRangesChineseFull describes Default + Japanese + full set of about 21000 CJK Unified Ideographs.
```
Is either of `GlyphRangesChineseFull` or `GlyphRangesChineseSimplifiedCommon` compatible with the previous `GlyphRangesChinese` ?
Because removing that function would demand a bump of major version, which I'd like to avoid. If there is a compatibility, this could be a forwarding function.

Galaco replied:
`GlyphRangesChinese` was renamed to `GlyphRangesChineseFull` in imgui 1.62. I've restored the original function for compatibility reasons, but added `GlyphRangesChineseFull`, and updated `GlyphRangesChinese` to instead call it. 

I wasn't 100% sure on that change, as it introduces an additional function call, but wanted to keep code duplication low and function naming conventions consistent with imgui's.

dertseha replied:
For these circumstances, the forwarding function is the best solution. It avoids a bump of major version, and Go provides options to mark such a function as "deprecated".
Also, the extra function call doesn't hurt. I'm sure there'd be some inlining, and this is an utility function called only once anyway.

dertseha commented on FontAtlas.go:
```diff
@@ -39,9 +39,14 @@ func (atlas FontAtlas) GlyphRangesJapanese() GlyphRanges {
+	return GlyphRanges(C.iggGetGlyphRangesChineseFull(atlas.handle()))
+}
+
+// GlyphRangesChineseSimplifiedCommon describes Default + Japanese + full set of about 21000 CJK Unified Ideographs.
```
What is the difference between `GlyphRangesChineseFull` and `GlyphRangesChineseSimplifiedCommon` if their comments are the same?

Galaco replied:
`GlyphRangesChinese` was renamed to `GlyphRangesChineseFull` in imgui 1.62. I've restored the original function for compatibility reasons, but added `GlyphRangesChineseFull`, and updated `GlyphRangesChinese` to instead call it. 

I wasn't 100% sure on that change, as it introduces an additional function call, but wanted to keep code duplication low and function naming conventions consistent with imgui's.

Galaco replied:
Apologies, fell into the copy-paste trap there. Comments for both have been updated to keep in line with imgui's comments

dertseha commented on WrapperConverter.h:
```diff
@@ -1,4 +1,4 @@
-#pragma once
+ #pragma once
```
Spurious whitespace change that is not needed :)

dertseha commented on imconfig.h:
```diff
@@ -13,19 +13,16 @@
 
 #pragma once
 
-extern "C" void iggAssert(int result, char const *expression, char const *file, int line);
```
removing `iggAssert` and the macro below removes functionality to capture assertions within ImGui in Go.
This should not be removed.

dertseha commented on imconfig.h:
```diff
@@ -13,19 +13,16 @@
 
 //---- Don't define obsolete functions/enums names. Consider enabling from time to time after updating to avoid using soon-to-be obsolete function/names.
-#define IMGUI_DISABLE_OBSOLETE_FUNCTIONS
+//#define IMGUI_DISABLE_OBSOLETE_FUNCTIONS
```
obsolete functions are to be disabled - this is intentional.

Galaco commented:
Afternoon. Above comments have been addressed:
* Removed whitespace from `WrapperConverter.h`
* Uncommented `#define IMGUI_DISABLE_OBSOLETE_FUNCTIONS` in `imconfig.h`
* Readded `iggAssert` to `imconfig.h`
* Corrected `GlyphRangesChineseSimplifiedCommon` & `GlyphRangesChineseFull` comments
* Readded `GlyphRangesChinese` function. It was renamed to `GlyphRangesChineseFull` in imgui 1.62. I wasn't 100% sure on preferred approach to organising thing here, so `GlyphRangesChineseFull` was added, and `GlyphRangesChinese` now calls it to maintain compatability.

dertseha commented:
Thank you for your changes!
I'll update some minor details on comments myself - I wanted to avoid another roundtrip just for that; I do like punctuation, so, function comments should probably end with a dot ;) I'll also add the deprecation mark to the old glyph atlas function.

After updating the readme (and a test with my own app) I'll finish it with a release.
